apps/src/code-studio/components/lessonExtras/LessonExtras.jsx

```diff
@@ -32,9 +32,18 @@ export default class LessonExtras extends React.Component {
       projectTypes,
       showLessonExtrasWarning
     } = this.props;
-    const nextMessage = /stage/.test(nextLevelPath)
-      ? i18n.extrasNextLesson({number: nextLessonNumber})
-      : i18n.extrasNextFinish();
+
+    let nextMessage = '';
+    switch (true) {
```

Comment:
This logic feels a bit fragile because it has to "match" the logic in `next_level_path_for_lesson_extras` (which calls the somewhat complicated `build_script_level_path` and depends on stuff in routes.rb).  I can't think of a great way to fix this though.  Maybe we can add comments to the unit tests that verify `next_level_path` that the logic here needs to be updated if the paths are ever updated?

Reply:
Yeah, I agree. I fixed the bug, but not the underlying fragility that caused it to regress in the first place. I'll add some comments.

Reply:
I'm curious about the switch statement choice here, I was thinking this could be written a little shorter as 
```
  let nextMessage = i18n.extrasNextUnitOverview();
  if(/lessons/.test(nextLevelPath)) {
    nextMessage = i18n.extrasNextLesson({number: nextLessonNumber});
  } else if (/congrats/.test(nextLevelPath)) {
    nextMessage = i18n.extrasNextFinish();
  }
```